benchmarking/mod/bblock/vdbench.py

```diff
@@ -175,33 +232,43 @@ def parse_benchmark_cases(self, testcase):
             testcase_dict["description"] = p[9]
         elif len(p) > 10:
             common.printout("ERROR","Too much columns found for test case ")
-	    sys.exit()
-	else:
+            sys.exit()
+        else:
             testcase_dict["description"] = ""
 
         return testcase_dict
 
     def run(self):
-	super(self.__class__, self).run()
+        super(self.__class__, self).run()
         user = self.cluster["user"]
-	nodes = self.cluster["nodes_distribution"]
-	waittime = 15
+        nodes = self.cluster["nodes_distribution"]
+        waittime = int(self.benchmark["runtime"]) + int(self.benchmark["rampup"])
+        dest_dir = self.cluster["tmp_dir"]
+        monitor_interval = self.cluster["monitoring_interval"]
         common.printout("LOG", "Start Running VdBench!")
-	common.pdsh(user, nodes, "cd %s; ./vdbench -f vdbench_test.cfg -o %s" % (self.cluster["bench_dir"], self.cluster["result_dir"]))
-	check_file = "%s/summary.html" % self.cluster["result_dir"]
-	self.check_run_success(check_file, 100)
+	common.pdsh(user, nodes, "date > %s/`hostname`_process_log.txt" % (dest_dir))
+        common.printout("LOG","Start system data collector under %s " % nodes)
+        common.pdsh(user, nodes, "top -c -b -d %s > %s/`hostname`_top.txt & echo `date +%s`' top start' >> %s/`hostname`_process_log.txt" % (monitor_interval, dest_dir, '%s', dest_dir))
+        common.pdsh(user, nodes, "mpstat -P ALL %s > %s/`hostname`_mpstat.txt & echo `date +%s`' mpstat start' >> %s/`hostname`_process_log.txt"  % (monitor_interval, dest_dir, '%s', dest_dir))
+        common.pdsh(user, nodes, "iostat -p -dxm %s > %s/`hostname`_iostat.txt & echo `date +%s`' iostat start' >> %s/`hostname`_process_log.txt" % (monitor_interval, dest_dir, '%s', dest_dir))
+        common.pdsh(user, nodes, "sar -A %s > %s/`hostname`_sar.txt & echo `date +%s`' sar start' >> %s/`hostname`_process_log.txt" % (monitor_interval, dest_dir, '%s', dest_dir))
+        common.pdsh(user, nodes, "cd %s; ./vdbench -f vdbench_test.cfg -o %s" % (self.cluster["bench_dir"], self.cluster["result_dir"]))
+       
+        self.chkpoint_to_log("vdbench start")       
+        check_file = "%s/summary.html" % self.cluster["result_dir"]
+        self.check_run_success(check_file, 100, "test")
         for wait in range(1, waittime):
             time.sleep(1)
```

Comment:
I think no need to sleep any more, since you have already checked the completed and slept in check_run_success() function.